89. Gray Code

```diff
@@ -0,0 +1,26 @@
+//Time: O (2 ^ n) Space: O(2 ^ n)
+class Solution {
+    public:
+        vector<int> grayCode(int n) 
+        {
+            if (n == 0 || n > 32) {
+                vector<int> result;
+                return result;
+            }
+            if (n == 1) {
+                vector<int> result;
+                result.push_back(0);
+                result.push_back(1);
+                return result;
+            }
+            vector<int> lastResult = grayCode(n - 1);
+            vector<int> result(1 << (n - 1), 0);
+            for (int i = 0; i < lastResult.size(); ++i) {
+                result[i] = lastResult[i];
+            }
+            for (int i = lastResult.size() - 1; i >= 0; --i) {
+                result[2 * lastResult.size() - i] = (1 << n) + lastResult[i];
```

Comment:
I don't think it is (1<<n). For example, if n = 2, 1<<2 = 4.

Reply:
Yes, it should be (1 << (n - 1)) here. I made the same mistake in line 17 and changed before submission, but somehow forgot to change this one.

Reply:
This is, really, the typical case I make mistakes. I have some ideas on algorithm but make some indexing error, variable name error etc when implement them. And when I have a test case on my mind I will think that the code run in the way I expected while it does not.

Reply:
Yes, very useful advice. I should use paper at the point when I am not too sure.